harness/determined/common/api/authentication.py

```diff
@@ -22,6 +23,23 @@ def salt_and_hash(password: str) -> str:
         return password
 
 
+def check_password_complexity(password: str) -> None:
```

Comment:
I feel like this is a suboptimal place to implement this logic if the goal is to keep requirements in the programatic front-end synchronized with requirements in the WebUI (as similarly the Typescript is a suboptimal place for the WebUI to hardcode its requirements).

I'd prefer to see it implemented only once and in master. I'm not concerned about the round-trip that must be taken with this strategy every time a password is checked.

Reply:
I agree, and we're tracking that (and considering trade-offs) with [DET-10209](https://hpe-aiatscale.atlassian.net/browse/DET-10209). Simplest is probably to do it all server-side and stop permitting client-side hashing, but I'm expecting that conversation to go the same way as discussions about minimum length, maximum length, etc. with many voices needing to be heard before we can say we're all aligned on a good solution.

Reply:
Nice, nice, nice. You were a step ahead of me.

What do you think about referencing that ticket in a TODO ([in Google style](https://google.github.io/styleguide/pyguide.html#312-todo-comments))? My motivation in a TODO would be to:

1. discourage later comers from putting too much work into this
2. give a reference to anyone who looks at this and has the same thought you and i did
3. help later DET-10209 workers find where they should clean things up as a part of their work.

Reply:
I have a slight preference for adding the TODO, but if you'd rather not, then it's not something I feel strongly about. Whatever you prefer, just want to get an explicit decision in order to close the comment.